lib/DependencyScan/ModuleDependencyCacheSerialization.cpp

```diff
@@ -1399,8 +1388,8 @@ void ModuleDependenciesCacheSerializer::writeImportStatementInfos(
   for (auto kind = ModuleDependencyKind::FirstKind;
        kind != ModuleDependencyKind::LastKind; ++kind) {
     auto modMap = cache.getDependenciesMap(kind);
-    for (const auto &entry : modMap) {
-      ModuleDependencyID moduleID = {entry.getKey().str(), kind};
+    for (const auto &entry : modMap.keys()) {
```

Comment:
Does the order matter here? Can you explain why it matters sometimes but not every time?

Reply:
It does not, and the prior change I had for sorting the keys was not actually needed, it was leftover debugging code I removed.